### PR TITLE
[DROOLS-7578] Evaluate Impact analysis for ansible integration rules

### PR DESCRIPTION
**Pull request by tkobayas**
- Quick prototype. Works for a basic example : ParserTest

Requires https://github.com/apache/incubator-kie-drools/pull/5594 to compile and run. `drools` and `drools-ansible-rulebook-integration` have different upstream repos ,so cross-repository build wouldn't work, hence this CI should fail until https://github.com/apache/incubator-kie-drools/pull/5594 is merged.

### CODE REVIEW

**Comment by tkobayas** [comment]
`ParserTest` generates a SVG file:
ParserTest-graph

**Comment by tkobayas** [comment]
Run CI after the 8.45.0-SNAPSHOT update is deployed.

File: .ci/pull-request-config.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@ version: "2.1"
 dependencies: ./project-dependencies.yaml
 
 build:
-  - project: kiegroup/drools
+  - project: apache/incubator-kie-drools
```
Comment (tkobayas) [inline .ci/pull-request-config.yaml:6]: Now changed from `kiegroup/drools` to `apache/incubator-kie-drools`.

File: pom.xml [hunk 2]
```diff
@@ -46,9 +47,9 @@
       </snapshots>
     </repository>
     <repository>
-      <id>jboss-public-repository-group</id>
-      <name>JBoss Public Repository Group</name>
-      <url>https://repository.jboss.org/nexus/content/groups/public/</url>
+      <id>apache-public-repository-group</id>
+      <name>Apache Public Repository Group</name>
+      <url>https://repository.apache.org/content/groups/public/</url>
```
Comment (tkobayas) [inline pom.xml:52]: Use apache repo rather than jboss repo even for SNAPSHOT

**Comment by tkobayas** [comment]
@mariofusco Please review, thanks!

**Comment by tkobayas** [comment]
Note that this visualization depends on https://github.com/apache/incubator-kie-drools/pull/5594 , so this doesn't run with drools `8.43.0.Final` which drools-ansible-rulebook-integration `1.0.x`branch currently depends on.

File: .ci/pull-request-config.yaml (hunk 1 above)
Comment (tkobayas) [reply .ci/pull-request-config.yaml:6]: Hi @lampajr , this is a little irregular case. `drools-ansible-rulebook-integration` is developed in `kiegroup`, but it depends on the latest SNAPSHOT of `apache/incubator-kie-drools`. Do you think these yaml changes are valid?

**Comment by tkobayas** [comment]
This seems to be expected checkout.
```
2023-12-11T07:53:20.4391431Z ##[group]Checkout summary
2023-12-11T07:53:20.4392442Z [INFO] [apache/incubator-kie-drools]
2023-12-11T07:53:20.4393919Z [INFO]      Project taken from apache/incubator-kie-drools:main (Branch HEAD 0abd1fcbb2c126d4a6b5ad67764bce9c894c4b10)
2023-12-11T07:53:20.4395379Z [INFO] [kiegroup/drools-ansible-rulebook-integration]
2023-12-11T07:53:20.4397264Z [INFO]      Project taken from kiegroup/drools-ansible-rulebook-integration:main (Branch HEAD 5ceabec812f72d10368a2e669b0d608da9d52bf8)
2023-12-11T07:53:20.4399070Z [INFO]      Merged tkobayas/drools-ansible-rulebook-integration:DROOLS-7578-ansible-visualization into branch main
2023-12-11T07:53:20.4400088Z ##[endgroup]
```

**Review by lampajr — APPROVED** [review]
CI changes look good to me

**Review by mariofusco — APPROVED** [review]

File: .ci/pull-request-config.yaml (hunk 1 above)
Comment (tkobayas) [reply .ci/pull-request-config.yaml:6]: Thank you for reviewing, @lampajr !

Comment (rgdoliveira) [reply .ci/pull-request-config.yaml:6]: @tkobayas I think this change is ok. But keep in mind that the idea is that we will have a kiegroup/drools main branch in sync with apache/incubator-kie-drools at some point (that will be our midstream). So maybe later we can move back to kiegroup/drools when that branch is in place?

Comment (mariofusco) [reply .ci/pull-request-config.yaml:6]: @rgdoliveira I wasn't aware that we wanted to restore kiegroup/drools and keep it in sync with the apache project (and tbh I don't see why we should), but anyway are you ok with this change for now? If so can you please approve it so I could merge? Thank you.

**Review by rgdoliveira — APPROVED** [review]

**Pull request merged**